src/Microsoft.Diagnostics.DebugServices/AnalyzeContext.cs

```diff
@@ -18,7 +18,7 @@ public AnalyzeContext()
         /// <summary>
         /// Current OS thread Id
         /// </summary>
-        public int CurrentThreadId { get; set; }
+        public uint? CurrentThreadId { get; set; }
```

Comment:
Yes, you are right.  l'll make it a ulong.

Reply:
But there are a lot of dbgeng APIs implemented in SOSHost that take uint like GetThreadContextById, GetCurrentThreadId, etc.  I can cast the ulong from AnalyzeContext.CurrentThreadId to uint in all the cases, but it means the native SOS code won't work for large macOS TIDs.

Reply:
If I change AnalyzeContext.CurrentThreadId to an int index (CurrentThreadIndex), there would be a lot less casts of the ulong OS TID. More of the dbgeng APIs in the hosting layer would work better because indexes are used most of the time. The ThreadInfo.ThreadId would also change to a ulong.

This is a pretty big change, but it is probably worth.

Reply:
It may not be worth it at all because the thread ids are limited to uint by the CLRMD IDataReader interface. I'm not to make any changes at this time maybe later.

We did have a short discussion for changing the thread id size in clrmd 2.0 but decided to keep it because Linux never creates any IDs greater than uint but we forgot/did know/think about MacOS.